### PR TITLE
Port some functions from OMZ Git

### PR DESCRIPTION
**Pull request by NovaDev94**
I ported some functions from OMZ Git plugin. Most of them follow the OMZ Git's conventions, except some conflicting with current abbreviations - or to fit current convention.

I split them into small commits, so feel free to reject some of them if you think they don't fit current convention.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,12 +128,15 @@ $ omf install https://github.com/jhillyerd/plugin-git
 | glr          | `git pull --rebase`                                  |
 | gp           | `git push`                                           |
 | gp!          | `git push --force-with-lease`                        |
+| gpo          | `git push origin`                                    |
 | gpv          | `git push --no-verify`                               |
 | gpv!         | `git push --no-verify --force-with-lease`            |
 | ggp          | push origin _current-branch_                         |
+| ggf          | `ggp --force`                                        |
+| ggfl         | `ggp --force-with-lease`                             |
```
Comment (jhillyerd) [inline README.md:136]: We've semi standardized on "force-with-lease", and using a `!` to indicate that.  So far I haven't had any problems using lease since we switched to it, never needed an full force.  I think if people really want to force it without a lease, they should type out `--force`.

Suggest we drop `ggf`, and rename `ggfl` to `gpo!` - what do you think?

File: README.md [hunk 2]
```diff
@@ -128,12 +128,15 @@ $ omf install https://github.com/jhillyerd/plugin-git
...
+| gpu          | `ggp --set-upsteam`                                  |
```
Comment (jhillyerd) [inline README.md:137]: `gpu` exists as a function already.  I'm not sure if the functionality is any different, as some of the functions are quite old and git built in options may have made them obsolete.

File: README.md [hunk 3]
```diff
@@ -158,23 +178,41 @@ $ omf install https://github.com/jhillyerd/plugin-git
...
+
+
 ### Everything Else
 
 | Abbreviation | Command                                                     |
 | ------------ | --------------------------------------------------------    |
 | g            | `git`                                                       |
-| ga           | `git add`                                                   |
-| gaa          | `git add --all`                                             |
-| gapa         | `git add --patch`                                           |
```
Comment (MartinDelille) [inline README.md:168]: I personnaly use this shortcut a lot! Please don't remove it or explain why you want to remove it!

Comment (MartinDelille) [reply README.md:168]: Oups sorry: didn't see you just put it in a specific section.

Comment (NovaDev94) [reply README.md:168]: Yeah, sorry the changes were a bit fragmented, so it's a bit hard to follow. I didn't remove or change any semantics of the defined abbreviations or functions in this PR (except `gpu` - which `--set-upstream` goes after the branch name - which is okay for most cases).

File: README.md (hunk 1 above)
Comment (NovaDev94) [reply README.md:136]: I'm good with it :D.

File: README.md (hunk 2 above)
Comment (NovaDev94) [reply README.md:137]: What I did was to make `gpu` an abbreviation instead (removed the function) since its definition is pretty much identical to `ggp` - which doesn't really need a dedicated function to serve that. Aliasing it to `ggp --set-upstream` should work for most cases (I'm well aware of the argument shifting, but still haven't been able to think up of a case where it fails). The only difference is `--set-upstream` is now pushed to the shifted - after the branch name.

```
git push --set-upstream origin (current_branch)
# versus
git push origin (current_branch) --set-upstream
```

File: README.md (hunk 1 above)
Comment (NovaDev94) [reply README.md:136]: Since `ggfl` is aliased to `ggp --force-with-lease`. And `gpo` is just `git push origin`, I think `ggp!` is more suitable - and also follow the convention.

I'm good with dropping `ggf`.

Comment (NovaDev94) [reply README.md]: Implemented here: https://github.com/jhillyerd/plugin-git/pull/39/commits/05a12235bfee675f8216b58f029af7e462e1dc0a

File: init.fish [hunk 4]
```diff
@@ -21,6 +21,7 @@ __git_abbr g          git
 __git_abbr ga         git add
 __git_abbr gaa        git add --all
 __git_abbr gapa       git add --patch
+__git_abbr gap        git patch
```
Comment (jhillyerd) [inline init.fish:24]: My git installs say patch is not a command, and I can't find online docs for this.

File: README.md [hunk 5]
```diff
@@ -161,14 +161,24 @@ $ omf install https://github.com/jhillyerd/plugin-git
 | gwip         | commit a work-in-progress branch                     |
 | gunwip       | uncommit the work-in-progress branch                 |
 
+### Files
```
Comment (jhillyerd) [inline README.md:190]:
```suggestion
### Local Files
```

Comment (NovaDev94) [reply README.md:190]: 👍

File: init.fish (hunk 4 above)
Comment (NovaDev94) [reply init.fish:24]: I just tried it again, it was a mistake on my end. Should by `git apply` instead.

**Comment by NovaDev94** [comment]
I'll squash the commits after reviews are done to avoid polluting the history.

**Comment by jhillyerd** [comment]
I think I've seen all the changes now.  Any other squashing that needs to be done?  I'd rather we don't squash the whole thing into a single commit, too many things changing at once.

**Comment by NovaDev94** [comment]
@jhillyerd Finished my squashing :D
